internal/cachecontroller/cache_controller.go

```diff
@@ -87,15 +90,20 @@ func (c *InMemoryCacheController) DetermineInvalidation(
 		return entry.LastModified
 	}
 
-	lastModified, err, _ := c.sf.Do(storeID, func() (interface{}, error) {
-		return c.findChangesAndInvalidate(ctx, storeID)
-	})
-
-	if err != nil {
-		return time.Time{}
-	}
-
-	return lastModified.(time.Time)
+	go func() {
```

Comment:
This would create a goroutine every time there is an expiration. Creating a goroutine seems more expensive than checking for an entry in a `sync.Map` which could be cleaned up _after_  `findChangesAndInvalidate` returns.

Reply:
updated to have go only for the invalidation path